### PR TITLE
fix(HB): IOS-1319 - Adding `SocketError` and error subscribing to `ExchangeInteractor`

### PR DESCRIPTION
**Pull request by thisisalexmcgregor**
## Objective

This is a progress PR for adding in some error handling in `SocketManager`. 

## Description

This PR includes the addition of `SocketError` as well as `ExchangeInteractor` subscribing to the error payloads returned by `SocketManager`. Note that this PR does not include UI updates in response to the returned errors. 

I also dialed back the logging a bit in `SocketManager`. I'm filtering out `Heartbeat` types as it was getting a bit noisy in the console log. 

## How to Test

1. Build and run
2. Go to the new exchange
3. Begin entering in some small values. You should see errors returned in the console log. e.g. `Decoded socket message of type SocketError`
4. Enter in a value that is within the appropriate limits (max/min). You should see the correct conversion. 

## Screenshot/Design assessment

N/A

## Merge Checklist

- [x] The PR uses a title supported by [.changelogrc](https://github.com/blockchain/My-Wallet-V3-iOS/blob/dev/.changelogrc#L6...L69).
- [x] Areas of technical debt are marked with a `// TICKET:` comment that includes a ticket number.
- [x] All unit tests pass.
- [ ] You have added unit tests.
- [x] New files added are within the correct directory. (e.g. if a file is required for unit tests to compile, be sure it is added to the tests target.)

### CODE REVIEW

File: Blockchain/Network/Models/SocketMessage.swift [hunk 1]
```diff
@@ -44,7 +44,9 @@ extension SocketMessageCodable {
         do {
             let decoded = try JSONType.decode(data: data)
             let socketMessage = SocketMessage(type: socketType, JSONMessage: decoded)
-            Logger.shared.debug("Decoded socket message of type \(JSONType.self)")
+            if JSONType.self != HeartBeat.self {
+                Logger.shared.debug("Decoded socket message of type \(JSONType.self)")
```
Comment (bchrisarriola) [inline Blockchain/Network/Models/SocketMessage.swift]: Just delete logging altogether? There's a ticket for improving logging with SocketManager (IOS-1325)

File: Blockchain/Network/SocketManager.swift [hunk 2]
```diff
@@ -23,6 +23,16 @@ class SocketManager {
...
         self.webSocketMessageSubject = PublishSubject<SocketMessage>()
     }
+    
+    /// Data providers should subscribe to this for error handling.
+    /// The reason we use this and not `webSocketMessageObservable` is
+    /// because since it's technically not an `Error` but rather a `SocketMessage`
+    /// the `onError` closure in the `webSocketMessageObservable` won't be called.
+    /// For simplicity sake, you'll want to observe both `webSocketMessageObservable`
+    /// and `webSocketErrorObservable`.
+    var webSocketErrorObservable: Observable<SocketMessage> {
```
Comment (bchrisarriola) [inline Blockchain/Network/SocketManager.swift]: Should this return `Observable<SocketError>` instead? So:

```
webSocketMessageObservable.filter {
    $0.type == .exchange && $0.JSONMessage is SocketError
}.map { message in
    return message.JSONMessage as! SocketError
}
```

File: Blockchain/Network/Models/SocketMessage.swift (hunk 1 above)
Comment (thisisalexmcgregor) [reply Blockchain/Network/Models/SocketMessage.swift]: Yep, I can do that. Wasn't sure if it was useful to anyone.

File: Blockchain/Network/SocketManager.swift (hunk 2 above)
Comment (thisisalexmcgregor) [reply Blockchain/Network/SocketManager.swift]: lol whoops this was actually dead code. Removed it. I'm doing what you described above already.

File: Blockchain/Network/Models/SocketMessage.swift (hunk 1 above)
Comment (thisisalexmcgregor) [reply Blockchain/Network/Models/SocketMessage.swift]: Done.

**Pull request merged**